pkg/controller/scyllacluster/resource.go

```diff
@@ -219,9 +221,25 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 	if err != nil {
 		return nil, fmt.Errorf("can't get rack labels: %w", err)
 	}
-	matchLabels := helpers.ShallowCopyMap(rackLabels)
+	_, rackOrdinal, ok := slices.Find(c.Spec.Datacenter.Racks, func(rack scyllav1.RackSpec) bool {
```

Comment:
nit: I suppose at this place you could just get the index as argument (the caller has it) but I don't feel strong

Reply:
done